packages/component-library-angular/src/checkbox/Checkbox.spec.ts

```diff
@@ -0,0 +1,396 @@
+
+      expect(handleChange).toHaveBeenCalledTimes(1);
+      expect(checkbox).toBeChecked();
+    });
+
+    it('can have a checked state in CSS', async () => {
+      const { container } = await render('<input checked="true" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+      });
+      const checkbox = container.querySelector(':checked');
+      expect(checkbox).toBeInTheDocument();
+    });
+  });
+
+  describe('invalid variant', () => {
+    it('renders a design system BEM modifier class name', async () => {
+      const { container } = await render('<input [invalid]="invalid" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+        componentProperties: {
+          invalid: true,
+        },
+      });
+
+      const checkbox = container.querySelector('input[type="checkbox"]');
+
+      expect(checkbox).toHaveClass('utrecht-checkbox--invalid');
+    });
+
+    it('is not invalid by default', async () => {
+      const { container } = await render('<input type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+      });
+
+      const checkbox = container.querySelector('input[type="checkbox"]');
+
+      expect(checkbox).not.toBeInvalid();
+    });
+
+    it('omits non-essential invalid attributes when not invalid', async () => {
+      const { container } = await render('<input type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+      });
+
+      const checkbox = container.querySelector('input[type="checkbox"]');
+
+      expect(checkbox).not.toHaveAttribute('aria-invalid');
+    });
+
+    it('can have an invalid state', async () => {
+      const { container } = await render('<input [invalid]="invalid" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+        componentProperties: {
+          invalid: true,
+        },
+      });
+
+      const checkbox = container.querySelector('input[type="checkbox"]');
+
+      expect(checkbox).toBeInvalid();
+    });
+
+    it('can have a invalid state in CSS', async () => {
+      const { container } = await render('<input [required]="required" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+        componentProperties: {
+          required: true,
+        },
+      });
+
+      const checkbox = container.querySelector(':invalid');
+
+      expect(checkbox).toBeInTheDocument();
+      expect(checkbox).toBeInvalid();
+    });
+  });
+
+  describe('disabled variant', () => {
+    it('renders a design system BEM modifier class name', async () => {
+      const { container } = await render('<input [disabled]="disabled" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+        componentProperties: {
+          disabled: true,
+        },
+      });
+
+      const checkbox = container.querySelector('input[type="checkbox"]:disabled');
```

Comment:
By using `:disabled` here, you are testing two things at once. I suggest to simply use `querySelector('.utrecht-checkbox')` here.